thingworx-import/Entities.xml

```diff
@@ -0,0 +1,885 @@
+            } else if (tag.type === 4) {
+                // Ewon String tag converts to Thingworx String
+                tagType = "STRING";
+            } else {
+                // Unknown tag type, default to Thingworx Number
+                logger.warn("Unknown tag type (" + tag.type + ") for tag: " + tag.name);
+                tagType = "NUMBER";
+            }
+        } else {
+            logger.warn("The connected Ewon device is not sending tag type information. Check that the Ewon Thingworx Connector has been updated on the Ewon device!");
+        }
+
+        // Add datapoint to Thing
+        me.InsertDataPoint({
+            tagName: tag.name /* STRING */ ,
+            ewonName: ewonName /* STRING */ ,
+            tagValue: tag.value /* STRING */ ,
+            timestamp: tag.timestamp /* STRING */ ,
+            tagQuality: tagQuality /* STRING */ ,
+            tagType: tagType /* STRING */ ,
+            isTalk2MDevice: false /* BOOLEAN */
+        });
+    }
+}
+run();
+
+// Registers that an update has been received
+me.ProcessTimeSinceUpdate({
+    hasUpdated: true
+});]]></code></Row></Rows></ConfigurationTable></ConfigurationTables></ServiceImplementation><ServiceImplementation description="" handlerName="Script" name="Talk2MSyncData"><ConfigurationTables><ConfigurationTable dataShapeName="" description="" isMultiRow="false" name="Script" ordinal="0"><DataShape><FieldDefinitions><FieldDefinition baseType="STRING" description="code" name="code" ordinal="0"/></FieldDefinitions></DataShape><Rows><Row><code><![CDATA[/*
+ * Talk2MSyncData gets and inserts all logged data from DataMailbox.
+ *
+ * @author HMS Networks, MU Americas Solution Center
+ */
+function parseEwon(ewon) {
+    // Gets Ewon name without leading/trailing spaces
+    var ewonName = ewon.name.trim();
+
+    // Loop through each tag listed
+    ewon.tags.forEach((tag) => {
+        var i;
+        if (tag.history) {
+            // Process each historical data point
+            tag.history.forEach((datapoint) => {
+
+                // Get tag data quality
+                var tagQuality = "GOOD";
+                if (datapoint.quality === "bad") {
+                    tagQuality = "BAD";
+                } else if (datapoint.quality === "unknown") {
+                    tagQuality = "UNKNOWN";
+                }
+
+                // Get tag data type
+                var tagType;
+                if (datapoint.dataType === "Int") {
+                    tagType = "INTEGER";
+                } else if (datapoint.dataType === "Bool") {
+                    tagType = "BOOLEAN";
+                } else if (datapoint.dataType === "Float") {
+                    tagType = "NUMBER";
+                } else if (datapoint.dataType === "UInt") {
+                    tagType = "NUMBER";
+                } else {
+                    logger.warn("Unknown data type (" + datapoint.dataType + ") for tag: " + tag.name);
+                    tagType = "NUMBER";
```

Comment:
Can these types be a (constant) variable? You are hand typing "NUMBER" in three places here.

Reply:
moved to constants